pallets/parachain-staking/src/lib.rs

```diff
@@ -1907,24 +1906,26 @@ pub mod pallet {
 		pub fn compute_top_candidates() -> Vec<T::AccountId> {
 			let mut candidates = <CandidatePool<T>>::get().0;
 			// order candidates by stake (least to greatest so requires `rev()`)
-			candidates.sort_unstable_by(|a, b| a.amount.partial_cmp(&b.amount).unwrap());
+			candidates.sort_by(|a, b| a.amount.partial_cmp(&b.amount).unwrap());
 			let top_n = <TotalSelected<T>>::get() as usize;
 			// choose the top TotalSelected qualified candidates, ordered by stake
-			candidates
+			let mut collators = candidates
 				.into_iter()
 				.rev()
 				.take(top_n)
 				.filter(|x| x.amount >= T::MinCollatorStk::get())
 				.map(|x| x.owner)
-				.collect::<Vec<T::AccountId>>()
+				.collect::<Vec<T::AccountId>>();
+			collators.sort();
+			collators
```

Comment:
So this is the core fix right? The sort we added sorts by collator id? That seems like a weird thing to do since they will all be in the set.

Reply:
Yes

> That seems like a weird thing to do since they will all be in the set.

`is_selected_candidate` uses binary search so that's why they're stored sorted by AccountId https://github.com/PureStake/moonbeam/blob/master/pallets/parachain-staking/src/lib.rs#L1658-L1660

Reply:
We should add a comment about why `sort()` exists here

Reply:
Oh, I see (from your other comment) -- we sort because we use `binary_search`, not because this needs to match some other (possibly incorrect) behavior.

Reply:
> match some other (possibly incorrect) behavior
That was my first thought. But I think the binary search point makes sense and addresses it.